pkg/detectlicense/licenses.go

```diff
@@ -45,14 +50,18 @@ var (
 	spdxTag = []byte("SPDX-License" + "-Identifier:")
 
 	spdxIdentifiers = map[string]License{
-		"Apache-2.0":   Apache2,
-		"BSD-1-Clause": BSD1,
-		"BSD-2-Clause": BSD2,
-		"BSD-3-Clause": BSD3,
-		"ISC":          ISC,
-		"MIT":          MIT,
-		"MPL-2.0":      MPL2,
-		"CC-BY-SA-4.0": CcBySa40,
+		"Apache-2.0":        Apache2,
+		"BSD-1-Clause":      BSD1,
+		"BSD-2-Clause":      BSD2,
+		"BSD-3-Clause":      BSD3,
+		"CC-BY-SA-4.0":      CcBySa40,
+		"GPL-3.0-only":      GPL3,
+		"ISC":               ISC,
+		"LGPL-2.1-or-later": LGPL21,
```

Comment:
I don't like that it's discarding the `-only` and `-or-later` information.

Reply:
That's a good point. When I was looking for the URL for LGPL21, I went into the license of one dependency (don't remember which one) to see if it was '-only' or '-or-later' and since it was '-or-later' I set LGPL21 to mean 'GNU Lesser General Public License v2.1 or later', however I didn't change the name.

So there are two things here:
1. I think I should change LGPL21's name (and probably the variable as well) to be correct and include the 'later' part.
2. Are all licenses that were being classified as LGPL21 of type 'or-later'? I checked one and went with the assumption all others were the same but if this may not be the case then I'll have to go over every dependency and figure out which one it is.

Reply:
@LukeShu I've update LGPL21 to LGPL21OrLater.
I think it's enough for now to know if we use GPL-like licenses or not, which we're not supposed to do on applications installed on customer servers.

Please let me know if you have any concerns.

Reply:
@LukeShu Just FYI, this is the PR where validation of GPL and AGPL will be done based on [this](https://www.notion.so/datawire/License-Management-5194ca50c9684ff4b301143806c92157) page.